### PR TITLE
added three functions for getting the pointer to value for maps, seqs…

### PR DESCRIPTION
**Pull request by ankushdesai**
… and tuple types

### CODE REVIEW

File: Src/Prt/Core/PrtValues.c [hunk 1]
```diff
@@ -697,7 +712,11 @@ PRT_VALUE *PrtMapUpdateHelper(_Inout_ PRT_VALUE *map, _In_ PRT_VALUE *key, _In_
 
 				PRT_VALUE *oldValue = next->value;
 				next->value = valueClone;
-				return oldValue;
+				if (oldValue != NULL)
+				{
+					PrtFreeValue(oldValue);
```
Comment (alexreinking) [inline Src/Prt/Core/PrtValues.c]: Running `PrtFreeValue` on NULL is a no-op

File: Src/Prt/Core/PrtValues.c [hunk 2]
```diff
@@ -813,6 +829,11 @@ void PRT_CALL_CONV PrtMapRemove(_Inout_ PRT_VALUE *map, _In_ PRT_VALUE *key)
 	}
 }
 
+PRT_VALUE ** PRT_CALL_CONV PrtMapGetLValue( _Inout_ PRT_VALUE *map, _In_ PRT_VALUE *key, _In_ PRT_BOOLEAN cloneKey)
+{
+	return PrtMapUpdateHelper(map, key, cloneKey, PrtMkNullValue(), PRT_FALSE);
```
Comment (alexreinking) [inline Src/Prt/Core/PrtValues.c]: The `PrtMkNullValue()` call here will leak memory when the key already exists. Maybe you mean `NULL` here?

File: Src/Prt/Core/PrtValues.c [hunk 3]
```diff
@@ -731,19 +750,16 @@ PRT_VALUE *PrtMapUpdateHelper(_Inout_ PRT_VALUE *map, _In_ PRT_VALUE *key, _In_
...
+	return &node->value;
 }
 
 
 void PRT_CALL_CONV PrtMapUpdateEx(_Inout_ PRT_VALUE *map, _In_ PRT_VALUE *key, _In_ PRT_BOOLEAN cloneKey, _In_ PRT_VALUE *value, _In_ PRT_BOOLEAN cloneValue)
 {
-	PRT_VALUE *oldValue = PrtMapUpdateHelper(map, key, cloneKey, value, cloneValue);
-	if (oldValue != NULL)
-	{
-		PrtFreeValue(oldValue);
```
Comment (alexreinking) [inline Src/Prt/Core/PrtValues.c]: Also a no-op. If statement not needed.

Comment (ankushdesai) [reply Src/Prt/Core/PrtValues.c]: Fixed.

**Comment by ankushdesai** [comment]
The two changes have been fixed.

**Pull request merged**